### DISCUSSION TITLE
What's the best way to allow other Repositorys ?

### DISCUSSION THREAD

**Discussion by vega82** [uyuni-project/uyuni#6662] [Q&A]
We use Uyuni in the latest Version (01.2023) to manage all of our Linux machines.
Some of these need some special repositorys (for example Google Chrome, R, Docker).
So what's the best way to make these repos available? Should i sync theses repos via custom channel to Uyuni and link these repos to those maschines? Is there a tutorial step by step how to do this? (I allready tried an failed).

Or should i enable these Internet repos for those Clients?
Because since 2022.11. all external repos are disabled by default during state apply via /usr/share/susemanager/salt/channels/disablelocalrepos.sls
Is there a way to allow explicit these repos?

**Comment by mcalmer** [comment]
Uyuni is central management server. So the idea is, that all repositories come from the Uyuni Server.
Create custom channels for your 3rd party repos, sync them and assign the channels to your system is the preferred way.

If you do not care about central management, you can also assign the repos directly and disable the automatic remove by setting the pillar `mgr_disable_local_repos: False`. During bootstrapping the system it will still remove the local repos to prevent conflicts with the software we install during this process. But after the bootstrapping this would not happen again and local repos are kept.
See also https://www.uyuni-project.org/uyuni-docs/en/uyuni/client-configuration/repositories.html

**Comment by vega82** [comment]
Hi, i don't get it. I tried to build a local docker repository and get this error:
2023/03/01 07:54:19 +02:00 Command: ['/usr/bin/spacewalk-repo-sync', '--channel', 'docker-ubuntu', '--type', 'deb', '--non-interactive']
2023/03/01 07:54:19 +02:00 Sync of channel started.
2023/03/01 07:54:19 +02:00 Plugin error: GPG verification failed: https://download.docker.com/linux/ubuntu/dists/jammy/InRelease

In the channel settings gpg check is disabled.
In my Docker Repo "Has Signed Metadata?" is also disabled.
What's wrong? 
I there a community driven solution for easy adding for the most popular repositorys?

**Comment by digdilem** [comment]
There is a different way to do this if you have relatively few machines using that repo, which is to use Uyuni to deploy .repo files to the client machines using a Configuration channel.  We do this for docker, pushing `/etc/yum.repos.d/docker-ce.repo` to those machines and then the client has the packages available for regular `yum install` and `yum update` operations.

This obviously doesn't give the benefits of repo mirroring that Uyuni does, but it can simplify life at small scale.
